Realm/RLMObject.h

```diff
@@ -122,194 +127,201 @@ NS_ASSUME_NONNULL_BEGIN
  
- If nested objects are included in the argument, `createInRealm:withValue:` will be called
+ If nested objects are included in the argument, `createInRealm:withValue:` will be recursively called
  on them.
  
- @param realm   The Realm in which this object is persisted.
- @param value   The value used to populate the object. This can be any key/value coding compliant
-                object, or a JSON object such as those returned from the methods in NSJSONSerialization, or
-                an NSArray with one object for each persisted property. An exception will be
-                thrown if any required properties are not present and no default is set.
-                
-                When passing in an NSArray, all properties must be present, valid and in the same order as the properties defined in the model.
+ @param realm   The Realm which should manage the newly-created object.
+ @param value   A value used to populate the object. This can be any key/value coding compliant
+                object, or a JSON object such as those returned from the methods in `NSJSONSerialization`, or
+                an `NSArray` with one object for each persisted property. An exception will be
+                thrown if any required properties are not present and and those properties were not defined with
+                default values.
+
+                When passing in an `NSArray`, all properties must be present, valid and in the same order
+                as the properties defined in the model.
  
- @see   defaultPropertyValues
+ @see   `defaultPropertyValues`
  */
 + (instancetype)createInRealm:(RLMRealm *)realm withValue:(id)value;
 
 /**
- Create or update an RLMObject in the default Realm with a given object.
+ Creates or updates a Realm object within the default Realm.
 
- This method can only be called on object types with a primary key defined. If there is already
- an object with the same primary key value in the default RLMRealm its values are updated and the object
- is returned. Otherwise this creates and populates a new instance of this object in the default Realm.
+ This method can only be called on Realm object types with a primary key defined. If there is already
+ an object with the same primary key value in the default Realm, its values are updated and the object
+ is returned. Otherwise, this method creates and populates a new instance of the object in the default Realm.
  
  If nested objects are included in the argument, `createOrUpdateInDefaultRealmWithValue:` will be
- called on them if have a primary key (`createInDefaultRealmWithValue:` otherwise).
+ recursively called on them if have a primary key, `createInDefaultRealmWithValue:` if they do not.
  
- This is a no-op if the argument is an RLMObject of the same type already backed by the target realm.
+ If there is a Realm object managed by the default
+
+ If the argument is a Realm object already managed by the default Realm, the argument's type is the same
```

Comment:
I would appreciate any opinions over how this sentence is worded. I think it still might not be entirely clear and/or accurate.